src/configureWorkspace/configurePython.ts

```diff
@@ -3,68 +3,224 @@
+        case "flask":
+            contents = flaskRequirements
+            break;
+        default:
+    }
+
+    return contents;
+}
+
+async function initializeForDebugging(context: ScaffolderContext, dockerfile: string, ports: number[], generateComposeFiles: boolean,
+                                      target: PythonFileTarget | PythonModuleTarget, projectType: PythonProjectType): Promise<void> {
+    const scaffoldContext: DockerDebugScaffoldContext = {
+        folder: context.folder,
+        platform: "python",
+        actionContext: context,
+        dockerfile: dockerfile,
+        generateComposeTask: generateComposeFiles,
+        ports: ports
+    }
+
+    const pyOptions: PythonScaffoldingOptions = {
+        target: target,
+        projectType: projectType
+    }
+
+    await dockerDebugScaffoldingProvider.initializePythonForDebugging(scaffoldContext, pyOptions);
+}
+
+export async function promptForLaunchFile(projectType?: PythonProjectType) : Promise<PythonFileTarget | PythonModuleTarget> {
+    const launchFilePrompt = defaultLaunchFile.get(projectType);
+
+    const opt: vscode.InputBoxOptions = {
+        placeHolder: launchFilePrompt.defaultFile,
+        prompt: launchFilePrompt.prompt,
+        value: launchFilePrompt.defaultFile,
+        validateInput: (value: string): string | undefined => { return value && value.trim().length > 0 ? undefined : "Enter a valid Python file path/module." }
+    };
+
+    const file = await ext.ui.showInputBox(opt);
+
+    // If the input has the .py extension, then assume it is a file.
+    // TODO: is there a more robust way to check?
+    if (file.toLowerCase().endsWith(".py")) {
+        return { file: file.replace(/\\/g, '/') };
+    } else {
+        return { module: file};
+    }
+}
+
+export async function scaffoldPython(context: ScaffolderContext): Promise<ScaffoldFile[]> {
+    const properties: TelemetryProperties & ConfigureTelemetryProperties = context.telemetry.properties;
+    const serviceName = context.folder.name;
+    const rootFolderPath: string = context.rootFolder;
+    const outputFolder = context.outputFolder ?? rootFolderPath;
+
+    const generateComposeFiles = await context.captureStep("compose", quickPickGenerateComposeFiles)();
+    const projectType = getPythonProjectType(context.platform);
+
+    const defaultPort = PythonDefaultPorts.get(projectType);
+    let ports = [];
+
+    if (defaultPort) {
+        ports = await context.promptForPorts([ defaultPort ]);
+    }
+
+    const launchFile = await context.captureStep("pythonFile", promptForLaunchFile)(projectType);
+    const dockerFileContents = genDockerFile(serviceName, launchFile, projectType, ports);
+
+    const files: ScaffoldFile[] = [
+        { fileName: 'Dockerfile', contents: dockerFileContents, open: true },
+        { fileName: '.dockerignore', contents: genCommonDockerIgnoreFile(context.platform) },
+        { fileName: 'requirements.txt', contents: genRequirementsFile(projectType) }
```

Comment:
Should we always generate this file?  If the user already has this file, isn't it likely it that it has the correct content for the project?  While we *do* prompt the user to overwrite any scaffolded file, that's under the assumption that we may be adding specific content for the purposes of the building/debugging flows and the user should be generally ok with overwriting it.  I'm not sure the same can be said of this file, though.

Reply:
That was one of the requirements, for covering the "new project and F5" scenario there has to be a requirements file. If the user happens to have one, then they would probably say "no" to our prompt of replacing it.

So from our side, instead of trying to get the actual project dependencies (i.e. do a "pip freeze" command), we agreed it is safer just to have a minimal list of requirements for the known frameworks, given we explicitly mention that file in the Dockerfile.

Reply:
Totally understand generating the file if it doesn't exist; I'm more concerned about the case where it's already there.  In that case, what we generate is probably not correct--that is, won't at least match what's already there--even if the user was ok with us overwriting it.

Reply:
I see your point, do you suggest we look if it exists, find if the minimum requirements are there and add them if not? It would be a good point to bring up in our meeting with Dan today since I'm sure he'll have valuable feedback.

Reply:
As a start, I'd just check whether it exists and, if so, *not* scaffold a new one, and not be concerned with its contents (for now).  That is, assume that, if it exists, it likely has all of the content needed.

Reply:
I believe only attempt to create it if it doesn't exist makes sense to me. If the user has a requirements.txt file then it would definitely more complete than the one we generate. 

I'll go ahead with that approach and bring it to discussion later with Uche.